SpatialGDK/Source/SpatialGDK/Private/Interop/RPCs/SpatialRPCService.cpp

```diff
@@ -0,0 +1,484 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "Interop/RPCs/SpatialRPCService.h"
+
+#include "EngineClasses/SpatialNetBitReader.h"
+#include "EngineClasses/SpatialPackageMapClient.h"
+#include "Interop/SpatialStaticComponentView.h"
+#include "SpatialConstants.h"
+#include "Utils/RepLayoutUtils.h"
+#include "Utils/SpatialLatencyTracer.h"
+
+DEFINE_LOG_CATEGORY(LogSpatialRPCService);
+
+namespace SpatialGDK
+{
+SpatialRPCService::SpatialRPCService(const ExtractRPCDelegate InExtractRPCCallback, const FSubView& InActorAuthSubView,
+									 const FSubView& InActorNonAuthSubView, USpatialNetDriver* InNetDriver, FTimerManager* InTimerManager)
+	: ExtractRPCCallback(InExtractRPCCallback)
+	, NetDriver(InNetDriver)
+	, TimerManager(InTimerManager)
+	, RPCStore(FRPCStore())
+	, ClientServerRPCs(InExtractRPCCallback, InActorAuthSubView, InNetDriver, RPCStore)
+	, MulticastRPCs(InExtractRPCCallback, InActorNonAuthSubView, RPCStore)
+	, AuthSubView(&InActorAuthSubView)
+{
+	PeriodicallyProcessIncomingRPCs();
+	IncomingRPCs.BindProcessingFunction(FProcessRPCDelegate::CreateRaw(this, &SpatialRPCService::ApplyRPC));
+}
+
+void SpatialRPCService::Advance()
+{
+	ClientServerRPCs.Advance();
+	MulticastRPCs.Advance();
+	// TODO: Cross server should go here too
```

Comment:
as discussed in slack. cross server RPCs will probably not go here for now as the refactor I'm doing gets rid of using TimerManager and the RPCContainer. It also doesn't need to use the RPCStore as the overall logic is a bit different due to using commands. So it's mainly about making sure we also pass in the subviews, I guess.

Reply:
removed the todo